corehq/motech/repeaters/tasks.py

```diff
@@ -139,43 +178,19 @@ def retry_process_repeat_record(repeat_record_id, domain):
     NOTE: Keep separate from process_repeat_record for monitoring purposes
     Domain is present here for domain tagging in datadog
     """
-    _process_repeat_record(RepeatRecord.objects.get(id=repeat_record_id))
+    return _process_repeat_record(RepeatRecord.objects.get(id=repeat_record_id))
 
 
 def _process_repeat_record(repeat_record):
-    if repeat_record.state == State.Cancelled:
```

Comment:
isn't this check still valid?

Reply:
(Assuming a repeater is only being processed by one worker at a time) this code can't be reached unless the repeat record is either pending or failed. `Repeater.repeat_records_ready` ensures that. (I'll add tests to show that.)

Reply:
> Just wondering about the lock and what it's purpose is

I discovered an important purpose for the lock. If `prefetch_multiplier` is not set to a small number (it is "1" on Production, and not set on Staging) then Celery will keep iterating repeaters, and queue up as many tasks as it thinks is reasonable, and it is not reasonable. On Staging (which has concurrency set to 4) it queued up 4,814,913 tasks. This blocked the queue for all but the domain with the most repeat records. Pausing data forwarding for the offending domain had no effect because the tasks were already queued, and so Celery continued to try to process its repeat records. (Purging the tasks in the queue fixed the problem.)

Reply:
I've put [the check back here](https://github.com/dimagi/commcare-hq/pull/34946/commits/5fe6d3c4c2c3c62a18f5edcfe007e04f78e6257d#diff-c8d8d8a27f8a60ae819abd6b25fc7fb8bf85bb6207a25c6ea3c1b3c0171b27a2R202).